### ISSUE TITLE
Compiler Error - Makefile:2162: //hal/phydm/phydm.mk: No such file or directory

### ISSUE THREAD

**Issue by ingejae** [tomaspinho/rtl8821ce#86]
Hello Tomas,

I tried to install the driver, downloading it from the url: 
_git clone https://github.com/tomaspinho/rtl8821ce.git_

and get the next error after running: **sudo make all**

_make ARCH=x86_64 CROSS_COMPILE= -C /lib/modules/4.15.0-64-generic/build M=/tmp/rtl8821ce  modules
make[1]: Entering directory '/usr/src/linux-headers-4.15.0-64-generic'
/tmp/rtl8821ce/Makefile:2162: //hal/phydm/phydm.mk: No such file or directory
/tmp/rtl8821ce/Makefile:2171: //rtl8821c.mk: No such file or directory
make[2]: *** No rule to make target '//rtl8821c.mk'.  Stop.
Makefile:1571: recipe for target '_module_/tmp/rtl8821ce' failed
make[1]: *** [_module_/tmp/rtl8821ce] Error 2
make[1]: Leaving directory '/usr/src/linux-headers-4.15.0-64-generic'
Makefile:2242: recipe for target 'modules' failed
make: *** [modules] Error 2_

I have a version of the dirver from August 28/2019 and works fine.

Have a nice day and thanks for the driver :-D

**Comment by sany98215** [comment]
I had update last version and got same error

**Comment by tomaspinho** [comment]
Why are you not installing via DKMS? There's a bunch of assumptions we make in the Makefile about the kind of installation that is being done.

**Comment by ingejae** [comment]
Hello,

I had to reinstall my notebook and followed the Issue-2 on next link:
_https://gist.github.com/debojyoti/228729498628e898497557db57619a28_

When I got the error, I remember that had a copy from the date mentioned above.

That copy from August 28/2019 worked fine.

**Comment by tomaspinho** [comment]
Try running this: https://github.com/tomaspinho/rtl8821ce#ubuntu--debian
And then that: https://github.com/tomaspinho/rtl8821ce#manual-installation-of-driver

**Comment by Electrocze** [comment]
@ingejae Hi can you please send me the version from August ?

**Comment by EvaneSnow** [comment]
@Electrocze If you make error, you can modify Makefile,
export TopDIR ?= the Makefile's path
for example:
export TopDIR ?= /home/snow/rtl8821ce
the two 'export TopDIR ?=' , you both modify

you can read the paper:
https://blog.csdn.net/qq_37116150/article/details/101611113

**Comment by Electrocze** [comment]
@EvaneSnow Okay thank you, because in this version i get in every linux os (sudo make all) error 2

**Comment by EvaneSnow** [comment]
@Electrocze not need sudo make all, you only make, it's ok

**Comment by Electrocze** [comment]
> @Electrocze not need sudo make all, you only make, it's ok

i try only make but same .

**Comment by EvaneSnow** [comment]
@Electrocze you can speak chinese

**Comment by EvaneSnow** [comment]
oh ,i know , do your system is ubuntu ?

**Comment by Electrocze** [comment]
> @Electrocze you can speak chinese

no i cant 
and now i try deepin

**Comment by EvaneSnow** [comment]
> > @Electrocze you can speak chinese
> 
> no i cant
> and now i try deepin

03:00.0 Network controller: Realtek Semiconductor Co., Ltd. RTL8821CE 802.11ac PCIe Wireless Network Adapter

do your wireless network adapter is 8821ce?
if isn't, the method doesn't work

I suggest you can read my article:
https://blog.csdn.net/qq_37116150/article/details/101611113

**Comment by Electrocze** [comment]
@EvaneSnow yeah i got rtl8821ce

**Comment by EvaneSnow** [comment]
> @EvaneSnow yeah i got rtl8821ce

en, first: git clone https://github.com/tomaspinho/rtl8821ce (you got)
cd cd rtl8821ce/

second:
gedit Makefile
and modife:
export TopDIR ?= 
for example:
ifneq (,$(findstring /usr/lib/dkms,$(PATH)))
    export TopDIR ?= /home/snow/rtl8821ce
else
    export TopDIR ?= /home/snow/rtl8821ce
endif

third:
make
sudo make install
sudo modprobe -a 8821ce

if the method doesn't work, I also hasn't others methods

**Comment by Electrocze** [comment]
@EvaneSnow thank you i will try :grin:

**Comment by Electrocze** [comment]
@EvaneSnow Still Error 2

**Comment by Staslend** [comment]
@EvaneSnow Ubuntu 18.04.3, kernel version 5.0.0, this helped me, thanks a lot!

**Comment by EvaneSnow** [comment]
> @EvaneSnow Ubuntu 18.04.3, kernel version 5.0.0, this helped me, thanks a lot!

不客气，能帮助你，我很开心

**Comment by EvaneSnow** [comment]
> @EvaneSnow Still Error 2

sorry, I also don't have others methods, but if your system is ubuntu, you can update your linux kernel version, and the version must be 14 or 14+;
and again try;
一般情况下，是不需要升级linux 内核的，但是你总是有错误，你可以试一下

https://blog.csdn.net/fljhm/article/details/79281655

**Comment by Electrocze** [comment]
@EvaneSnow it's okay i have older version and working fine:)

**Comment by ingejae** [comment]
Hello All,

Did you fix the issue?

If not, tell me where I can publish the older version I have downloaded.

Remember, this installs the Wifi driver for Lenovo 330S 15ARR (rtl8821ce)

I'm using Xubuntu 16.04

**Comment by Electrocze** [comment]
@ingejae probably no , it will be nice to send your version

**Comment by ingejae** [comment]
you can download it from the next link.

It will be available for a few days.

http://116.202.111.144/rtl8821ce.zip

**Comment by Electrocze** [comment]
@ingejae thanks :)

**Comment by Innovictor** [comment]
> > @EvaneSnow Still Error 2
> 
> sorry, I also don't have others methods, but if your system is ubuntu, you can update your linux kernel version, and the version must be 14 or 14+;
> and again try;
> 一般情况下，是不需要升级linux 内核的，但是你总是有错误，你可以试一下
> 
> https://blog.csdn.net/fljhm/article/details/79281655

**Hello @snow, I followed your story to CSDN and used Google Translate. Thank you so much for writing this. I am responding now with Wifi from my HP X360 convertible. Your method works after I've been on this for 2 days.**

**Comment by philippludwig** [comment]
> There's a bunch of assumptions

Are these documented somewhere?

**Comment by tomaspinho** [comment]
@philippludwig no, but it's mostly due to the `TopDIR` variable in here and how we derive the root path of the source tree.

**Comment by tomaspinho** [comment]
If you need to build this in-kernel, you can take a look at https://github.com/endlessm/linux/tree/master/drivers/net/wireless/rtl8821ce

**Comment by philippludwig** [comment]
Thanks. My issue is that I don’t have `dkms` available on my system, so I tried to compile the driver manually against the kernel source tree. However, I did not have any success and now decided that I will just get a different wifi adapter.
